### ISSUE TITLE
token can be used several times

### ISSUE THREAD

**Issue by individual-it** [owncloud/twofactor_totp#59]
1. login using a token
2. remember the token
3. logout
4. login again using the same token

From [RFC6238](https://tools.ietf.org/html/rfc6238) :
```
   Note that a prover may send the same OTP inside a given time-step
   window multiple times to a verifier.  The verifier MUST NOT accept
   the second attempt of the OTP after the successful validation has
   been issued for the first OTP, which ensures one-time only use of an
   OTP.
```

**Comment by PVince81** [comment]
likely not a regression as this part of the code wasn't touched

**Comment by karakayasemi** [comment]
If someone listens to the user-server conversation and learns the secret, he/she still should be unable to log on. I am planning to add a column to app table for the last secret and prevent this situation.

**Comment by PVince81** [comment]
@karakayasemi ^

**Comment by PVince81** [comment]
- [ ] revert on release branch: https://github.com/owncloud/twofactor_totp/pull/64
- [ ] revert on master ?

**Comment by karakayasemi** [comment]
Somehow mapper update is not working in this column:https://github.com/owncloud/twofactor_totp/blob/master/lib/Service/Totp.php#L123. lastValidatedKey column always null. I will debug it.

**Comment by karakayasemi** [comment]
I found my mistake. To prevent method does not exist warnings in createMock function, I implemented set function for last_validated_key column in here: https://github.com/owncloud/twofactor_totp/blob/master/lib/Db/TotpSecret.php#L65. Obviously, entity setters are doing more than just assignment. They are also updating UpdatedFields of the entity. I will create a fix for my mistake. 
Should I open PR for master or which way I should follow?
